mbuild/compound.py

```diff
@@ -190,6 +191,52 @@ def periodicity(self):
     def periodicity(self, periods):
         self._periodicity = np.array(periods)
 
+    def view_hierarchy(self, show_ports=False):
+        """View a Compounds hierarchy of compounds as a chart. """
+        try:
+            import networkx as nx
+        except ImportError:
+            raise ImportError('Ensure Networkx is installed correctly')
+        try:
+            import matplotlib.pyplot as plt
+        except ImportError:
+            raise ImportError('Ensure matplotlib is installed correctly')
+
+        from networkx.readwrite import json_graph
+        from mbuild.utils.visualization import d3_tree_template
+
+        compound_tree = nx.DiGraph()
+        compound_tree.add_node(self.kind)
+        compound_frequency = Counter([self.kind])
+        for sub_compound in self._yield_parts(Compound):
+            if not show_ports and sub_compound.kind in ["Port", "subport"]:
+                continue
+            compound_frequency[sub_compound.kind] += 1
+            compound_tree.add_node(sub_compound.kind)
+            if sub_compound.parent:
+                compound_tree.add_edge(sub_compound.parent.kind, sub_compound.kind)
+        labels = {"'children'": '"children"', "'name'": '"name"'}
+        for compound in compound_tree:
+            node_key = "'{}'".format(compound)
+            labels[node_key] = '"{} {:d}"'.format(compound, compound_frequency[compound])
+        # This code is taken from: https://gist.github.com/mbostock/4339083
+        # The directed graph is converted to a json format and manipulated to be
+        # correctly displayed
+        # The graph is visualized in html by d3.json visualization
```

Comment:
Minor quibble: this should read "d3.js" which, here, stands for javascript.

I would maybe move all this to the doc string and write:
```
"""Visualize a compound hierarchy as a tree.

A tree is constructed from the compound hierarchy with self as the root. The tree is then rendered in a web browser window using D3.js.

Note
------
Portions of this code are adapted from https://gist.github.com/mbostock/4339083.